core/local/layers/linux.js

```diff
@@ -0,0 +1,180 @@
+    }
+  }
+
+  async process (events /*: AtomWatcherEvent[] */) {
+    // TODO logger
+    // TODO ignore
+    const batch /*: LayerEvent[] */ = []
+    for (const event of events) {
+      switch (event.action) {
+        case 'created':
+          const eAdd = await this.buildAddEvent(event.path)
+          batch.push(eAdd)
+          if (eAdd.doc.docType === 'folder') {
+            this.watch(eAdd.doc.path)
+          }
+          break
+        case 'modified':
+          batch.push(await this.buildUpdateEvent(event.path))
+          break
+        case 'deleted':
+          const eDel = await this.buildRemoveEvent(event.path, event.kind)
+          batch.push(eDel)
+          const w = this.watchers.get(eDel.doc.path)
+          if (w) {
+            w.dispose()
+            this.watchers.delete(eDel.doc.path)
+          }
+          break
+        case 'renamed':
+          batch.push(await this.buildMoveEvent(event.path, event.oldPath))
+          break
+        default:
+          throw new Error(`Unknown atom/watcher action ${event.action}`)
+      }
+    }
+    this.next.process(batch)
+  }
+
+  async buildAddEvent (abspath /*: string */) /*: Promise<LayerAddEvent> */ {
+    let doc /*: ?Metadata */
+    const fpath = this.relativePath(abspath)
+    const stats = await fse.stat(abspath)
+    if (stats != null && stats.isDirectory()) {
+      doc = buildDir(fpath, stats)
+    } else {
+      doc = buildFile(fpath, stats, '')
+    }
+    return { action: 'add', abspath, doc }
+  }
+
+  async buildUpdateEvent (abspath /*: string */) /*: Promise<LayerUpdateEvent> */ {
+    let doc /*: ?Metadata */
+    const fpath = this.relativePath(abspath)
+    const stats = await fse.stat(abspath)
+    if (stats != null && stats.isDirectory()) {
+      doc = buildDir(fpath, stats)
+    } else {
+      doc = buildFile(fpath, stats, '')
+    }
+    return { action: 'update', abspath, doc }
+  }
+
+  async buildRemoveEvent (abspath /*: string */, kind /*: string */) /*: Promise<LayerRemoveEvent> */ {
+    const fpath = this.relativePath(abspath)
+    let doc /*: ?Metadata */
+    if (kind === 'directory') {
+      doc = buildDir(fpath, new fs.Stats())
+    } else {
+      doc = buildFile(fpath, new fs.Stats(), '')
+    }
+    return { action: 'remove', abspath, doc }
+  }
+
+  async buildMoveEvent (abspath /*: string */, oldpath /*: string */) /*: Promise<LayerMoveEvent> */ {
+    let doc /*: ?Metadata */
+    let src /*: ?Metadata */
+    const fpath = this.relativePath(abspath)
+    const stats = await fse.stat(abspath)
+    if (stats != null && stats.isDirectory()) {
+      doc = buildDir(fpath, stats)
+      src = buildDir(oldpath, new fs.Stats())
+    } else {
+      doc = buildFile(fpath, stats, '')
+      src = buildFile(oldpath, new fs.Stats(), '')
+    }
+    return { action: 'move', abspath, doc, src }
+  }
+
+  relativePath (abspath /*: string */) {
+    return path.relative(this.syncPath, abspath)
```

Comment:
Will we have to compute relative paths on other platforms too?

Reply:
I think we will have to. Why?